21.1.5 update (#2)
DevExpress-Examples/wpf-docklayoutmanager-serialize-docklayoutmanager-when-tabbeddocumentuiservice-is-used: Readme.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T155653)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# WPF Dock Layout Manager - Serialize DockLayoutManager When You Use the TabbedDocumentUIService

> Some of DockLayoutManager’s panel groups can contain documents that were created from ViewModel with [IDocumentManagerService](https://docs.devexpress.com/WPF/18171/mvvm-framework/services/predefined-set/document-services). To serialize and restore them correctly, it is necessary to perform the following steps.

Reviewer: `To serialize and restore them correctly, it is necessary to perform the following steps.`

What is "them"? Does it refer to the panel groups or the documents?

Author: ```suggestion
Some of DockLayoutManager’s panel groups can contain documents that were created from ViewModel with [IDocumentManagerService](https://docs.devexpress.com/WPF/18171/mvvm-framework/services/predefined-set/document-services). To serialize and restore these panels and their child documents correctly, it is necessary to perform the following steps.

```

> Item names are used to identify items when saving/restoring the layout. For this reason, it will be necessary to set unique names for all DockLayoutManager panels. In the current scenario, this can be done by setting the DocumentPanel.[BindableName](https://docs.devexpress.com/WPF/DevExpress.Xpf.Docking.BaseLayoutItem.BindableName) property in the [TabbedDocumentUIService.DocumentPanelStyle](https://docs.devexpress.com/WPF/DevExpress.Xpf.Docking.TabbedDocumentUIService.DocumentPanelStyle).

Reviewer: ```suggestion
Item names are used to identify items when saving/restoring the layout. For this reason, it is necessary to set unique names for all DockLayoutManager panels. In the current scenario, this can be done by setting the DocumentPanel.[BindableName](https://docs.devexpress.com/WPF/DevExpress.Xpf.Docking.BaseLayoutItem.BindableName) property in the [TabbedDocumentUIService.DocumentPanelStyle](https://docs.devexpress.com/WPF/DevExpress.Xpf.Docking.TabbedDocumentUIService.DocumentPanelStyle).
```

> Another important point is that the DockLayoutManager's saving/restoring mechanism is not the [XamlWriter](https://docs.microsoft.com/en-us/dotnet/api/system.windows.markup.xamlwriter?redirectedfrom=MSDN&view=windowsdesktop-6.0) alternative - it does not save/restore content of its panels. It will be necessary to additionally restore it. In this example, we used the following approach:

Reviewer: ```suggestion
Another important point is that the DockLayoutManager's save/restore mechanism is not the [XamlWriter](https://docs.microsoft.com/en-us/dotnet/api/system.windows.markup.xamlwriter?redirectedfrom=MSDN&view=windowsdesktop-6.0) alternative — it does not save/restore the content of its panels. It is necessary to restore it. In this example, we use the following approach:
```
